src/components/NewCommentForm.tsx

```diff
@@ -45,7 +123,14 @@ export const NewCommentForm: React.FC = () => {
             name="email"
             id="comment-author-email"
             placeholder="email@test.com"
-            className="input is-danger"
+            className={classNames(
+              'input',
+              {
+                'is-danger': (emailError && !email),
+              },
+            )}
+            value={email}
+            onChange={({ target }) => setEmailValue(target.value)}
```

Comment:
Better not use anonimous functions in JSX.
Please, create for it function in component.
And check other places